source/common/router/config_impl.h

```diff
@@ -201,9 +201,17 @@ class HashPolicyImpl : public HashPolicy {
   Optional<uint64_t> generateHash(const std::string& downstream_addr,
                                   const Http::HeaderMap& headers) const override;
 
+  class HashImpl {
```

Comment:
If this is a pure interface, it probably belongs in `include/envoy/router`. Also, for conventions, we tend not to suffix pure interfaces with `Impl`.

Reply:
My thought was that it's part of HashPolicyImpl's implementation, not part of its publicly accessible API. I'll change "HashImpl" to "HashMethod"; the intent was to signify that this is an interface for various ways of producing a hash.